syft/core/utils.py

```diff
@@ -1,6 +1,97 @@
+                                     torch.CharTensor,
+                                     torch.ShortTensor,
+                                     torch.IntTensor,
+                                     torch.LongTensor])
+
+    def encode(self, obj, retrieve_tensorvar=None):
+        if retrieve_tensorvar is not None:
+            self.retrieve_tensorvar = retrieve_tensorvar
+        if self.retrieve_tensorvar:
+            return (self.python_encode(obj), self.found_tensorvar)
+        else:
+            return self.python_encode(obj)
+
+    def python_encode(self, obj):
+        if isinstance(obj, (int, float, str)) or obj is None:
+            return obj
+        elif isinstance(obj, self.tensorvar_types):
+            if self.retrieve_tensorvar:
+                self.found_tensorvar.append(obj)
+            key = '__'+type(obj).__name__+'__'
+            return { key: '_fl.{}'.format(obj.id) }
+        elif isinstance(obj, (tuple, set, bytearray)):
+            key = '__'+type(obj).__name__+'__'
+            return {key:[self.python_encode(i) for i in obj]}
+        elif isinstance(obj, list):
+            return [self.python_encode(i) for i in obj]
+        elif isinstance(obj, dict):
+            return {
+                k: self.python_encode(v)
+                for k, v in obj.items()
+            }
+        else:
+            try:
+                return { '__eval__': str(obj) }
+            except:
+                print('Unknown type', type(obj))
+            return None
+
+class PythonJSONDecoder(json.JSONDecoder):
+    """
+        Decode JSON and replace python types when need
+        Retrieve Torch objects replaced by their id
+    """
+    def __init__(self, worker, *args, **kwargs):
+        super(PythonJSONDecoder, self).__init__(*args,
+            object_hook=self.custom_obj_hook, **kwargs)
+        self.worker = worker
+        self.tensorvar_types = tuple([torch.autograd.Variable,
+                                     torch.nn.Parameter,
+                                     torch.FloatTensor,
+                                     torch.DoubleTensor,
+                                     torch.HalfTensor,
+                                     torch.ByteTensor,
+                                     torch.CharTensor,
+                                     torch.ShortTensor,
+                                     torch.IntTensor,
+                                     torch.LongTensor])
+
+    def custom_obj_hook(self, dct):
+        pat = re.compile('__(.+)__')
+        for key, obj in dct.items():
+            try:
+                obj_type = pat.search(key).group(1)
+                if obj_type in map(lambda x: x.__name__, self.tensorvar_types):
+                    pattern_var = re.compile('_fl.(.*)')
+                    id = int(pattern_var.search(obj).group(1))
+                    return self.worker.get_obj(id)
+                elif obj_type in ('tuple', 'set', 'bytearray', 'eval'):
```

Comment:
This line is a big problem -- eval on eval can for sure execute arbitrary code

Reply:
Yes sure ! But this is not the last commit, it is the previous one. On the last commit, there is a strong focus on limiting the things we eval